keymanager-lib/src/methods.rs

```diff
@@ -9,23 +9,30 @@ use oasis_core_runtime::{
     enclave_rpc::Context as RpcContext,
 };
 
-/// Maximum age of ephemeral key in the number of epochs.
+/// Maximum age of an ephemeral key in the number of epochs.
 const MAX_EPHEMERAL_KEY_AGE: EpochTime = 10;
+/// Maximum age of a fresh height in the number of blocks.
+///
+/// A height is considered fresh if it is not more than specified amount
+/// of blocks lower than the height of the latest trust root.
+const MAX_FRESH_HEIGHT_AGE: u64 = 50;
 
 /// See `Kdf::get_or_create_keys`.
 pub fn get_or_create_keys(req: &LongTermKeyRequest, ctx: &mut RpcContext) -> Result<KeyPair> {
     authorize_private_key_generation(&req.runtime_id, ctx)?;
+    validate_height_freshness(req.height, ctx)?;
 
     Kdf::global().get_or_create_keys(req)
 }
 
 /// See `Kdf::get_public_key`.
 pub fn get_public_key(
     req: &LongTermKeyRequest,
-    _ctx: &mut RpcContext,
+    ctx: &mut RpcContext,
 ) -> Result<Option<SignedPublicKey>> {
     // No authentication or authorization.
     // Absolutely anyone is allowed to query public long-term keys.
+    validate_height_freshness(req.height, ctx)?;
```

Comment:
Probably not needed for public key queries.

Reply:
I though it was needed, as even though public keys are available to anyone, the key manager would (sometimes) like to communicate only with runtimes that have a fresh enough state.

Took comment into account and removed validation in both public queries.